pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -171,9 +191,24 @@ pub mod pallet {
 	/// This can only be modified by an admin.
 	#[pallet::storage]
 	#[pallet::getter(fn relayer)]
-	pub(crate) type RelayerList<T: Config> =
+	pub type RelayerList<T: Config> =
 		StorageDoubleMap<_, Blake2_128Concat, Domain, Blake2_128Concat, DomainAddress, ()>;
 
+	/// Storage for outbound messages that will be processed during the
+	/// `on_idle` hook.
+	#[pallet::storage]
+	#[pallet::getter(fn outbound_message_queue)]
+	pub type OutboundMessageQueue<T: Config> = StorageNMap<
+		_,
+		(
+			NMapKey<Blake2_128Concat, Domain>,
+			NMapKey<Blake2_128Concat, T::AccountId>,
+			NMapKey<Blake2_128Concat, T::Message>,
```

Comment:
This means, the same message will overwrite past once, but it should be possible to send the same message twice. We need ids unfortunately for the queue.

Reply:
Added a nonce for keeping track of each outbound/failed message.